### ISSUE TITLE
Add Branch Normalisation when calling /output json

### ISSUE THREAD

**Issue by gep13** [ParticularLabs/GitVersion#272]
As discussed in this closed PR: https://github.com/Particular/GitVersion/pull/271

I believe that it would make sense to allow the ability to perform branch normalisation when calling GitVersion with ```/output json```.  Currently, within my psake script, I have to call GitVersion twice, once with ```/output buildserver``` to perform the branch normalisation, and set the build number, and then again with ```/output json``` so I can get the GitVersion variables to use later within my psake script.

I believe that allowing something like:

```/output json /normalisebranches```

would be a great additional switch to add to GitVersion to prevent the need to call it twice.

If this flag addition is not liked, then I believe that at the very least, when calling ```/output buildserver``` the json output should also be included, so that the variables can be parsed and used.

**Comment by pvandervelde** [comment]
Has there been any progress on this?

**Comment by gep13** [comment]
> Has there been any progress on this?

To the best of my knowledge, no.

GitVersion is currently under-going a fairly major refactor at the moment to allow it to better handle different versioning strategies and support for both CI and CD usages.  This will see  GitVersion bumped to a new major version.  Once this work is complete, we should be in a position to tackle this.

**Comment by JakeGinnivan** [comment]
ok, so the work is done.

What needs to be done to fix this issue? Does someone want to submit a PR to the release/3.0.0 branch to enable normalisation by default.

Should it even be opt in? We need to do it by default most of the time right?

**Comment by JakeGinnivan** [comment]
@gep13 can you check this against master. 

We will only perform normalisation if running on a build server, but the original issue of it not running if the output was json should be fixed.

If we need to normalise for builds not on the build server then lets create a new issue.

**Comment by gep13** [comment]
@JakeGinnivan so I am confused...

If I run:

```
/output json
```

Does branch normalisation happen or not?

**Comment by JakeGinnivan** [comment]
It will on the build server..

`/output buildserver` is misleading.. It basically allows gitversion to output variables etc to the build server. We always detect if running on the build server though and normalise

**Comment by gep13** [comment]
Oh really, that part I didn't know.  I thought the inspection for whether running on a CI Server only happened with that flag was passed in.

Good to know!  In which case, I will give this a try when I get a minute (on a new project that isn't using TeamCity just now, but will get back to it sooner, rather than later).

**Comment by JakeGinnivan** [comment]
For reference, new logic is much easier to follow:

https://github.com/ParticularLabs/GitVersion/blob/master/GitVersionCore/ExecuteCore.cs#L13
https://github.com/ParticularLabs/GitVersion/blob/master/GitVersionCore/GitPreparer.cs#L34

**Comment by gep13** [comment]
Noice! :+1:
